rmp-serde/src/decode.rs

```diff
@@ -283,6 +292,93 @@ fn read_u32<R: Read>(rd: &mut R) -> Result<u32, Error> {
     rd.read_u32::<byteorder::BigEndian>().map_err(Error::InvalidDataRead)
 }
 
+#[derive(Debug)]
+struct ExtDeserializer<'a, R: 'a> {
+    rd: &'a mut R,
+    len: u32,
+    read_tag: bool,
```

Comment:
If you want, I think replacing these two variables with an enum

```
enum ExtDeserializerState {
    None,
    ReadTag,
    ReadBinary
}
```

could be slightly cleaner and byte-efficient. Would let us use `match` statements in the various places, and only have 1 byte stored rather than 2.

But the current approach isn't that bad either - and it's arguably simpler. Your choice!

Reply:
enum state machine looks good, it's more "rusty" :joy: